mlflow/server/js/src/experiment-tracking/pages/prompts/utils.ts

```diff
@@ -112,3 +132,149 @@ export const buildSearchFilterClause = (searchFilter?: string): string => {
     return `name ILIKE '%${searchFilter}%'`;
   }
 };
+
+/**
+ * Parse model config from tag value (JSON string).
+ * Returns undefined if tag doesn't exist or JSON parsing fails.
+ */
+export const getModelConfigFromTags = (tags?: KeyValueEntity[]): PromptModelConfig | undefined => {
+  const configTag = tags?.find((tag) => tag.key === PROMPT_MODEL_CONFIG_TAG_KEY);
+  if (!configTag?.value) {
+    return undefined;
+  }
+
+  try {
+    return JSON.parse(configTag.value) as PromptModelConfig;
+  } catch (error) {
+    console.error('Failed to parse model config:', error);
+    return undefined;
+  }
+};
+
+/**
+ * Convert form data to backend model config format.
+ * Returns undefined if no fields have values.
+ */
+export const formDataToModelConfig = (formData: PromptModelConfigFormData): PromptModelConfig | undefined => {
+  const hasAnyValue = Object.values(formData).some((v) => !!v);
+  if (!hasAnyValue) {
+    return undefined;
+  }
+
+  const config: PromptModelConfig = {};
+
+  if (formData.provider?.trim()) {
+    config.provider = formData.provider.trim();
+  }
```

Comment:
super nit but we can do like 
```suggestion
  const provider = formData.provider?.trim();
  if (provider) {
    config.provider = provider;
  }
```

Reply:
done!